libs/npm/core/src/hooks/api/utils/mergeChartSettings.ts

```diff
@@ -0,0 +1,163 @@
+import { IChartSectionSettingsModel } from '..';
+import { defaultChartSectionSettings } from '../constants';
+
+/**
+ * Generates a new chart settings configuration based on the default, current and overridden values.
+ * @param defaultChartOptions Original chart options provided by the template. Only used if 'currentSettings' is undefined.
+ * @param currentSettings The current chart settings you want to merge.
+ * @param overrideSettings The chart settings to apply and override the current chart settings
+ * @returns New chart settings.
+ */
+export const mergeChartSettings = (
```

Comment:
We now configure the chart template options when creating the report.  This may break older reports.